### PR TITLE
Add deprecated method warn when called by user 

### PR DESCRIPTION
**Pull request by YczYanchengzhe**
## What is the purpose of the change

Add deprecated method warn when called by user , related with #10850

## Checklist
- [x] Make sure there is a [GitHub_issue](https://github.com/apache/dubbo/issues) field for the change (usually before you start working on it). Trivial changes like typos do not require a GitHub issue. Your pull request should address just this issue, without pulling in other changes - one PR resolves one issue.
- [ ] Each commit in the pull request should have a meaningful subject line and body.
- [ ] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [ ] Check if is necessary to patch to Dubbo 3 if you are work on Dubbo 2.7
- [ ] Write necessary unit-test to verify your logic correction, more mock a little better when cross module dependency exist. If the new feature or significant change is committed, please remember to add sample in [dubbo samples](https://github.com/apache/dubbo-samples) project.
- [ ] Add some description to [dubbo-website](https://github.com/apache/dubbo-website) project if you are requesting to add a feature.
- [ ] GitHub Actions works fine on your own branch.
- [ ] If this contribution is large, please follow the [Software Donation Guide](https://github.com/apache/dubbo/wiki/Software-donation-guide).

### CODE REVIEW

File: dubbo-common/src/main/java/org/apache/dubbo/common/constants/LoggerCodeConstants.java [hunk 1]
```diff
@@ -230,6 +230,8 @@ public interface LoggerCodeConstants {
 
     String CLUSTER_FAILED_GROUP_MERGE = "2-20";
 
+    String CLUSTER_METHOD_DEPRECATED = "2-21";
+
```
Comment (AlbumenJ) [inline dubbo-common/src/main/java/org/apache/dubbo/common/constants/LoggerCodeConstants.java:234]: Please submit a PR related with this code in apache/dubbo-website.

**Review by AlbumenJ — CHANGES_REQUESTED** [review]

**Comment by AlbumenJ** [comment]
As for https://github.com/apache/dubbo/issues/10850, we want to check all of the methods annotated with `@Deprecated`.

**Comment by YczYanchengzhe** [comment]
I have several doubts when I am looking for an abandoned method

1. How to find abandoned methods? At present, I am looking for all methods that have added @ Deprecated annotation. I find that there are a lot of data that meet the conditions. Is there a better way?

2. For methods with @ Deprecated, do you need to add logs? If the class is abandoned, do you need to add logs for all methods of the class. For abstract class, final class, and tool class, do you need to add warn logs? For some abandoned get/set methods, do you need to add warn logs?

**Comment by AlbumenJ** [comment]
> I have several doubts when I am looking for an abandoned method
> 
> 1. How to find abandoned methods? At present, I am looking for all methods that have added @ Deprecated annotation. I find that there are a lot of data that meet the conditions. Is there a better way?
> 2. For methods with @ Deprecated, do you need to add logs? If the class is abandoned, do you need to add logs for all methods of the class. For abstract class, final class, and tool class, do you need to add warn logs? For some abandoned get/set methods, do you need to add warn logs?

Is it possible to generate some code when compiling to achieve this?

**Pull request closed without merging**